static/js/util/util.js

```diff
@@ -180,6 +181,31 @@ export function makeProfileProgressDisplay(active: string) {
   </svg>;
 }
 
+/**
+ * Determine the lesser of the current step or the first incomplete step
+ */
+export function currentOrFirstIncompleteStep(currentStep: ?string,
+  validatedStep: ?string): string {
+
+  // a null validated step means all steps were valid
+  // see lib/validation/profile.js#validateProfileComplete
+  if (R.isNil(validatedStep)) {
+    validatedStep = R.last(PROFILE_STEP_ORDER);
+  }
+
+  if (R.isNil(currentStep)) {
+    currentStep = validatedStep;
+  }
+
+  const currentStepIndex = R.indexOf(currentStep, PROFILE_STEP_ORDER);
+  let stepIndex = R.indexOf(validatedStep, PROFILE_STEP_ORDER);
+
+  // take the lesser of the page we're on and the least completed step
+  const idealStepIndex = Math.min(stepIndex, currentStepIndex);
+
+  return PROFILE_STEP_ORDER[idealStepIndex];
+}
```

Comment:
Good idea, still need to manually test, but this handles the nulls and passes my unit tests:

```es6
const getStepIndices = R.map(R.indexOf(R.__, PROFILE_STEP_ORDER));
const lastStep = R.last(PROFILE_STEP_ORDER);

export const currentOrFirstIncompleteStep = R.compose(
    R.nth(R.__, PROFILE_STEP_ORDER),
    R.reduce(R.min, Infinity),
    getStepIndices,
    R.append(lastStep),
    R.reject(R.isNil),
    R.pair
);
```